modules/menu/static_menu.c

```diff
@@ -556,7 +557,15 @@ static int dial_handler(struct re_printf *pf, void *arg)
 	}
 
 	re_hprintf(pf, "call uri: %s\n", uri);
-	err = ua_connect(ua, &call, NULL, uri, VIDMODE_ON);
+
+	const char ud_sentinel[] = "userdata=";
+	char *ud_pos = strstr(carg->prm, ud_sentinel);
+	char *user_data = NULL;
+	if (ud_pos != NULL) {
+		user_data = ud_pos + strlen(ud_sentinel);
+	}
```

Comment:
In baresip `re_regex` is often used for parsing. This is much more powerful and should simplify the code. Maybe we should extend the function `menu_ua_carg()` to parse also `userdata=xxx`.

Reply:
I've spent some time trying to find a regex for it. The best that I could come up with is this: 

`([^ ]+) ?([1234567890]+)? ?(userdata=[^ ]+)?`

which I tested here https://regex101.com with the following inputs:

```
+123456789
+123456789 0
+123456789 0 userdata=somedata
+123456789 userdata=somedata
```
Unfortunately, the `re_regex` function doesn't seem to behave the same.